messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+        }
+
+        if (logger.isDebugEnabled()) {
+            Optional<Cause> optionalCause = letter.cause();
+            if (optionalCause.isPresent()) {
+                logger.debug("Adding dead letter [{}] because [{}].", letter.message(), optionalCause.get());
+            } else {
+                logger.debug("Adding dead letter [{}] because the sequence identifier [{}] is already present.",
+                             letter.message(), sequence);
+            }
+        }
+
+        DeadLetterEntryMessage converter = converters
+                .stream()
+                .filter(c -> c.canConvert(letter.message()))
+                .findFirst()
+                .map(c -> c.toEntry(letter.message(), serializer))
+                .orElseThrow(() -> new IllegalArgumentException("No converter found"));
+
+        transactionManager.executeInTransaction(() -> {
+            Long maxIndexForQueueIdentifier = getNextIndexForSequence(sequence);
+            DeadLetterEntry deadLetter = new DeadLetterEntry(processingGroup,
+                                                             sequence,
+                                                             maxIndexForQueueIdentifier,
+                                                             converter,
+                                                             letter.enqueuedAt(),
+                                                             letter.lastTouched(),
+                                                             letter.cause().orElse(null),
+                                                             letter.diagnostics(),
+                                                             serializer);
+            entityManager().persist(deadLetter);
+        });
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void evict(DeadLetter<? extends M> letter) {
+        if (!(letter instanceof JpaDeadLetter)) {
+            throw new WrongDeadLetterTypeException(
+                    "Evict should be called with a JpaDeadLetter instance. Instead got: [" + letter.getClass().getName()
+                            + "]");
+        }
+        JpaDeadLetter<M> jpaDeadLetter = (JpaDeadLetter<M>) letter;
+
+        transactionManager.executeInTransaction(
+                () -> entityManager().createQuery("delete from DeadLetterEntry dl where dl.deadLetterId=:deadLetterId")
+                                     .setParameter("deadLetterId", jpaDeadLetter.getId())
+                                     .executeUpdate());
+    }
+
+    private EntityManager entityManager() {
+        return entityManagerProvider.getEntityManager();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void requeue(@Nonnull DeadLetter<? extends M> letter,
+                        @Nonnull Function<DeadLetter<? extends M>, DeadLetter<? extends M>> letterUpdater)
+            throws NoSuchDeadLetterException {
+        if (!(letter instanceof JpaDeadLetter)) {
```

Comment:
I would expect a test case in the `JpaSequencedDeadLetterQueueTest` class for this check.